src/routes.js

```diff
@@ -44,4 +45,20 @@ router.post('/coffeegrams', async ctx => {
   ctx.redirect(`/users/${ctx.state.currentUser.username}`);
 });
 
+router.post('/users/:id', async ctx => {
+  var { bio } = ctx.request.body;
+  if (ctx.params.id !== ctx.state.currentUser.id) {
+    ctx.request.flash('error', 'You\'re not allowed to edit someone else\'s profile');
+    return ctx.redirect('back');
+  }
+  if (isBlank(bio)) {
+    ctx.request.flash('error', 'Your bio looks empty!');
+    return ctx.redirect('back');
+  }
+  var id = ctx.params.id;
```

Comment:
Oh wait, lies. I see that you are using the user's ID when linking to this route. Just FYI, it's not a good ID to mix and match like this. It's safer to always link to a user by ID, or by username.